libcaf_core/caf/flow/op/retry.test.cpp

```diff
@@ -0,0 +1,197 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/fixture/flow.hpp"
+#include "caf/test/scenario.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/log/test.hpp"
+
+using namespace caf;
+
+namespace {
+
+struct fixture : test::fixture::deterministic, test::fixture::flow {
+  // Similar to retry::subscribe, but returns a retry_sub pointer instead of
+  // type-erasing it into a disposable.
+  template <class T = int, class Predicate>
+  auto raw_sub(caf::flow::observable<int> in, caf::flow::observer<int> out,
+               Predicate predicate) {
+    using sub_t = caf::flow::op::retry_sub<T, Predicate>;
+    auto ptr = make_counted<sub_t>(coordinator(), out, std::move(predicate),
+                                   in);
+    out.on_subscribe(caf::flow::subscription{ptr});
+    return ptr;
+  }
+};
+
+WITH_FIXTURE(fixture) {
+
+SCENARIO("the retry operator retries observable multiple times") {
+  int retry_count = 3;
```

Comment:
Counting down seems odd, since the variable doesn't actually count the retries (despite its name). I'd initialize with 0 and add a `max_retry_count` constant. By actually counting up, we can also use the `retry_count` for the tests below.